pkg/controller/subscription/subscription_controller.go

```diff
@@ -101,13 +101,16 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 		return reconcile.Result{}, nil
 	}
 
-	logrus.Infof("RHMI upgrade available")
-
 	latestRHMIInstallPlan, err := rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.client)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	if latestRHMIInstallPlan.Spec.Approved {
```

Comment:
The upgrade is already approved, move on.

Reply:
The above function `rhmiConfigs.IsUpgradeAvailable()` should be checking if there is an upgrade available. Is there a timing issue here where it will report an upgrade and the installplan is approved?

Reply:
The problem seems to be that the installplan can be approved = true, but not yet processing, as all we are interested in is approving the installplan, checking on that field seemed more sensible.

Or in other words, once the installplan is approved, we have nothing left to do, so let OLM get on with it's thing now.